ksp_plugin_test/plugin_integration_test.cpp

```diff
@@ -0,0 +1,634 @@
+﻿#include "ksp_plugin/plugin.hpp"
+
+#include <algorithm>
+#include <limits>
+#include <vector>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "testing_utilities/solar_system.hpp"
+
+namespace principia {
+
+using quantities::Abs;
+using quantities::ArcTan;
+using quantities::Cos;
+using quantities::Sin;
+using quantities::Sqrt;
+using si::Day;
+using si::Hour;
+using si::Minute;
+using si::Radian;
+using si::AstronomicalUnit;
+using testing_utilities::SolarSystem;
+using testing_utilities::ICRFJ2000Ecliptic;
+using ::testing::AllOf;
+using ::testing::Eq;
+using ::testing::Ge;
+using ::testing::Gt;
+using ::testing::Le;
+using ::testing::Lt;
+using testing_utilities::AbsoluteError;
+using testing_utilities::RelativeError;
+
+namespace ksp_plugin {
+
+class PluginIntegrationTest : public testing::Test {
+ protected:
+  PluginIntegrationTest()
+      : icrf_to_barycentric_positions_(ICRFJ2000Ecliptic::origin,
+                                       Barycentric::origin,
+                                       ircf_to_barycentric_linear_),
+        looking_glass_(Permutation<ICRFJ2000Ecliptic, AliceSun>::XZY),
+        solar_system_(SolarSystem::AtСпутник1Launch(
+                      SolarSystem::Accuracy::kAllBodiesAndOblateness)),
+        initial_time_(42 * Second),
+        planetarium_rotation_(1 * Radian),
+        plugin_(make_not_null_unique<Plugin>(
+                    initial_time_,
+                    planetarium_rotation_)),
+        bodies_(solar_system_->massive_bodies()) {
+    sun_gravitational_parameter_ =
+        bodies_[SolarSystem::kSun]->gravitational_parameter();
+    satellite_initial_displacement_ =
+        Displacement<AliceSun>({3111.0 * Kilo(Metre),
+                                4400.0 * Kilo(Metre),
+                                3810.0 * Kilo(Metre)});
+    auto const tangent =
+        satellite_initial_displacement_ * Bivector<double, AliceSun>({1, 2, 3});
+    Vector<double, AliceSun> const unit_tangent = Normalize(tangent);
+    EXPECT_THAT(
+        InnerProduct(unit_tangent,
+                     satellite_initial_displacement_ /
+                         satellite_initial_displacement_.Norm()),
```

Comment:
That way we get the cosine of the angle, more meaningful than comparing the projected length with 0 metres (and normalizes the error).